src/main/scala/akka/actor/BeamLoggingFSM.scala

```diff
@@ -50,6 +64,20 @@ trait BeamLoggingFSM[S, D] extends FSM[S, D] { this: Actor =>
 
     if (debugEvent && oldState != newState)
       log.debug("###FSM-transition### actor:" + self.toString() + " transition: " + oldState + " -> " + newState)
+
+    if (debugMessages) {
+      val (tick, triggerId) = currentTickAndTriggerId
+      val msg = BeamFSMMessage(context.sender(), context.self, event, tick, triggerId)
+      context.system.eventStream.publish(msg)
+    }
+  }
+
+  private def currentTickAndTriggerId: (Int, Long) = {
+    this match {
+      case x: HasTickAndTrigger =>
+        (x.getCurrentTick.getOrElse(-1), x.getCurrentTriggerIdOrGenerate)
+      case _ => (-9999, -9999L) //shouldn't happen because all the FSM actors HasTickAndTrigger
```

Comment:
Should we maybe throw instead? That way we find and fix any bug?

Reply:
I changed to -1, -1. It just means that there's no trigger id. It will be replaced by the id from the incoming payload. I don't want to force every future FSM actor to implement HasTickAndTrigger.

Reply:
@wrashid It might be worth forcing this (or maybe logging as a warning at least) as the point of the task is to make sure we can build out the actor flow at any point?